baby-bear/src/baby_bear.rs

```diff
@@ -153,6 +154,36 @@ impl Field for BabyBear {
 
         Some(x2013265919)
     }
+
+    // We hard code computing the 7'th root for rescue.
+    fn exp_root(&self, _power: u64) -> Self {
```

Comment:
It seems a bit risky to override it like this, in case different powers are used later. How about overriding `exp_u64` itself, but in a conditional way? Something like

```rust
fn exp_u64(&self, power: u64) -> Self {
    if power == 1725656503 {
        // Apply the more optimal addition chain for 1725656503.
    } else {
        // Fall back to the default impl.
        // Unfortunately Rust doesn't have syntax for calling a default impl,
        // so we'd need to expose the default logic as a global function,
        // fn exp_u64_by_squaring<F: Field>(x: F, power: u64) -> F { ... }
    }
}
```

Hopefully the branch will be optimized away if the compiler inlines it with a constant `power`.

Reply:
Sure, I'd be happy to switch to that option. I guess I was thinking that if we used different powers we could add conditional branching to exp_root.

The only downsides I can see is that we may want/need to keep "1725656503" (and a similar variable for the other fields) somewhere as a constant as having to recompute it regularly might be rather expensive. Also if the branch isn't optimized away it will slow down all our exponentials.